core/init.go

```diff
@@ -146,76 +163,108 @@ func initOauth2ForService(config *tomlConfig, url string, c *http.Client) {
 		} else {
-			s := fmt.Sprintf("Failed to set up jwt authentication with errorcode %d.", resp.StatusCode)
+			s := fmt.Sprintf("Failed to set up oauth2 authentication with errorcode %d.", resp.StatusCode)
 			lc.Error(s)
 		}
 	}
 }
 
-func initKongRoutes(url string, c *http.Client, r *KongRoute, path string, name string) {
-	req, err := sling.New().Base(url).Post(path).BodyJSON(r).Request()
+func initJWTAuth(config *tomlConfig, url string, c *http.Client) {
+	jwtParams := &KongJWTPlugin{
+		Name: config.KongAuth.Name,
+	}
+
+	req, err := sling.New().Base(url).Post(PluginsPath).BodyForm(jwtParams).Request()
 	resp, err := c.Do(req)
 	if err != nil {
-		s := fmt.Sprintf("Failed to set up routes for %s with error %s.", name, err.Error())
+		s := fmt.Sprintf("Failed to set up jwt authentication.")
+		defer resp.Body.Close()
 		lc.Error(s)
 	} else {
 		if resp.StatusCode == 200 || resp.StatusCode == 201 || resp.StatusCode == 409 {
-			lc.Info(fmt.Sprintf("Successful to set up route for %s.", name))
+			lc.Info("Successful to set up jwt authentication")
 		} else {
-			s := fmt.Sprintf("Failed to set up route for %s with error %s.", name, resp.Status)
+			s := fmt.Sprintf("Failed to set up jwt authentication with errorcode %d.", resp.StatusCode)
 			lc.Error(s)
 		}
 	}
 }
 
-//redirect request for 8001 to an admin service of 8000, and add authentication
-func initKongAdminInterface(config *tomlConfig, url string, c *http.Client) {
-	adminServiceParams := &KongService{
-		Name:     "admin",
-		Host:     config.KongURL.Server,
-		Port:     config.KongURL.AdminPort,
-		Protocol: "http",
+func initJWTAuthForRoute(config *tomlConfig, url string, c *http.Client, service string) {
+	jwtParams := &KongJWTPlugin{
+		Name: config.KongAuth.Name,
 	}
-	req, err := sling.New().Base(url).Post(ServicesPath).BodyForm(adminServiceParams).Request()
+
+	req, err := sling.New().Base(url + "routes/" + service + "/").Post(PluginsPath).BodyForm(jwtParams).Request()
 	resp, err := c.Do(req)
 	if err != nil {
-		s := fmt.Sprintf("Failed to set up service for admin loopback with error %s.", err.Error())
+		s := fmt.Sprintf("Failed to set up jwt authentication.")
+		defer resp.Body.Close()
 		lc.Error(s)
 	} else {
 		if resp.StatusCode == 200 || resp.StatusCode == 201 || resp.StatusCode == 409 {
-			lc.Info("Successful to set up admin loopback.")
+			lc.Info("Successful to set up jwt authentication")
 		} else {
-			lc.Error("Failed to set up admin loopback.")
+			s := fmt.Sprintf("Failed to set up jwt authentication with errorcode %d.", resp.StatusCode)
+			lc.Error(s)
 		}
 	}
+}
 
-	adminRouteParams := &KongRoute{Paths: []string{"/admin"}}
-	adminRoutePath := fmt.Sprintf("%sadmin/routes", ServicesPath)
-	req, err = sling.New().Base(url).Post(adminRoutePath).BodyJSON(adminRouteParams).Request()
-	resp, err = c.Do(req)
+func initKongRoutes(url string, c *http.Client, r *KongRoute, name string) {
+	routesubpath := "services/" + name + "/routes"
+	lc.Info(routesubpath)
```

Comment:
this message should be more descriptive about what it is printing off to the logs